### ISSUE TITLE
Boost version compatibility

### ISSUE THREAD

**Issue by onavratil-monetplus** [hazelcast/hazelcast-cpp-client#707]
https://github.com/hazelcast/hazelcast-cpp-client/blob/2de5385101a19a2da0dd24c60341d898fd759288/CMakeLists.txt#L145

the `find_package` command has `find_package(<PackageName> [version] ....)` syntax. The referenced *Latest bootstrap versions* are often not packaged and need custom compilation/install, yet this is silently ignored by the build and later leads to more cryptic compilation issues. 

Adding the `[version]` parameter can clarify this issue.

**Comment by ihsandemir** [comment]
@onavratil-monetplus I did not understand what you mean in this issue. What exactly is the problem? Currently, without the version it will try find the boost available and will try to work with it. What do you mean by `Latest bootstrap versions` ?

**Comment by ihsandemir** [comment]
@onavratil-monetplus We will put back the version. It was there before as 1.72 as the minimum but we need to first explore what is the smallest version we can support given the current code base and put the lowest possible version number, 1.72 is really high for a lot of systems default installations.

**Comment by ihsandemir** [comment]
I just researched what would be the minimum version we can put considering our code base. I tried versions starting with 1.52 and here are my findings:

1. Can not be less than 1.65 since `boost::atomic_shared_ptr` was introduced by this version.
2. Endian library: `endian_load, endian_store` came with 1.71. We can do work around solutions (
[workaround_load_store_endian.txt](https://github.com/hazelcast/hazelcast-cpp-client/files/5780851/workaround_load_store_endian.txt)) like using `boost::endian::endian_load<boost::uint16_t, 2, boost::endian::order::big>(*)` but it does not look very good.

Hence, I think that it is best to keep 1.72 as the boost minimum required version in the library.
